### PR TITLE
Refactor Step 3

### PR DESCRIPTION
**Pull request by heliocastro**
Big refactor on Object Types
Components realigned

### CODE REVIEW

**Comment by heliocastro** [comment]
@hoangnt2 Big one this one, it took me some hours.
Somehow i broke the authentication somewhere, and if you can use your skills to spot error, i appreciate

**Comment by hoangnt2** [comment]
@heliocastro sure, let me check it

**Comment by hoangnt2** [comment]
@heliocastro I don't see any error, could you show me where you got the authentication error?

**Comment by heliocastro** [comment]
@hoangnt2  Locally i can't login anymore with my regular email/password 
Did something changed on sw360 that i need recompile ?

**Comment by hoangnt2** [comment]
@heliocastro I can login normally, i guess maybe you are missing oauth client in sw360

**Comment by hoangnt2** [comment]
@heliocastro, By the way, I got this error on the vulnerability page. When I replace the Link component from next-intl/link with next/link the error goes away. Do you know the reason?
image

File: messages/en.json [hunk 1]
```diff
@@ -632,7 +632,7 @@
         "The new Component and new Release will be created, do you want to import?": "The new Component and new Release will be created, do you want to import?",
         "This release contains": "This release <strong>{name}</strong> contains: ",
         "This currently only supports SPDX RDF/XML files with a uniq described top level node": "This currently only supports SPDX RDF/XML files with a uniq described top level node",
-        "The release has been successfully linked to project": "The release <strong>{releaseName}</strong> has been successfully linked to project <strong>{projectName}</strong>",
+        "The release has been successfully linked to project": "The release <strong>{releaseName}</strong> has been successfully linked to project <str7ong>{projectName}</strong>",
```
[the remark this answers is not in the document]
Comment (heliocastro) [reply messages/en.json]: Done

**Comment by heliocastro** [comment]
> @heliocastro, By the way, I got this error on the vulnerability page. When I replace the Link component from next-intl/link with next/link the error goes away. Do you know the reason? image

Yes, of course.
netx-intl depends on "use-client'
But even after adding it, fails continues, because the Table entries
Inside the table we have React objects as a React component created dynamically per row, that think itself as a server component.

In this case, we are passing a client component inside a server component, and then, the warinings
```javascript
    const columns = [
        {
            id: 'vulnerabilities.externalId',
            name: t('External Id'),
            formatter: (externalId: string) =>
                _(
                    <>
                        <Link href={`/vulnerabilities/detail/${externalId}`} className='text-link'>
                            {externalId}
                        </Link>
                    </>
                ),
            sort: true,
        },
```

**Comment by hoangnt2** [comment]
@heliocastro, Oh thanks I got it

**Comment by heliocastro** [comment]
@hoangnt2 Merged, so this should be officially last breaking change.
From now on, i will do more atomic commits. 
Is working on my side the auth now

**Pull request merged**